### PR TITLE
6618 audiocom redirect tagging

### LINKED ISSUE #6618: Add campaign info to audio.com requests
Add information to those requests so that we know where they originate from in the Audacity UI.

### PR DESCRIPTION
**Pull request by saintmatthieu**
Resolves: #6618

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

## QA

When redirected to audio.com, the actions below should lead to the URL opened in the browser ending with
`mtm_campaign=Audacity&mtm_source=<audacity version>&mtm_content=<see below>`,
e.g. `https://audio.com/matthieu-hodgkinson/audio/french-break-136-bpm-pl/edit?mtm_campaign=Audacity&mtm_source=Audacity-3.6.0-alpha-20240610&mtm_content=Share_Audio_Button`
Note that this should also apply to the "link-account" button associated with these actions, as well as to the 
"Visit audio.com" button in case the max number of projects was exceeded.

### Example
I want to save my project to the cloud,
image

but there have to link my account: 
image

It leads me to audio.com and the link has `Save_to_Cloud_Menu` appended:
image

Once I have linked my account, upload begins and then I see the success dialog, and clicking on "view my project" leads me to 
image, which also has that `Save_to_Cloud_Menu` appended.

I save another project, and say I reached my limit. I see 
image
and clicking "Visit audio.com" leads me to my projects page, 
image, and the link also has `Save_to_Cloud_Menu` appended.

### Actions (to be checked by QA)

| Action                            | string                            | notes |
| --------------------------------- | --------------------------------- |---|
|image | `Share_Audio_Button`                | |
| image                  | `Share_Audio_Menu`                  | |
| image           | `Share_Audio_Export_Menu`           | |
| image| `Share_Audio_Export_Extra_Menu`     | mp3 or any other format but not "selected audio" |
| image               | `Save_to_Cloud_Menu`                | |
| image  | `Save_Project_Save_to_Cloud_Menu`   | and then when the modal opens, use the cloud option |
| image                       | `Prefs_Panel`                       |  |
| image   | `Update_Cloud_Audio_Preview_Menu`   | |
| image   | `Link_Audiocom_Account_Help_Menu`   | If done successfully, there shouldn't be a redirect. If it fails, though, then there should, and the string should be in the URL. One way of failing it is to give some random token. (There are several other reasons why this might fail but testing them all would probably be an overkill. |
| image              | Open_From_Cloud_Menu              | |

### CODE REVIEW

File: modules/sharing/mod-cloud-audiocom/CloudProjectOpenUtils.cpp [hunk 1]
```diff
@@ -156,13 +156,15 @@ AudacityProject* OpenProjectFromCloud(
...
+   auto authResult =
+      PerformBlockingAuth(potentialTarget, AudiocomTrace::OpenFromCloudMenu);
 
    if (authResult.Result != AuthResult::Status::Authorised)
    {
       LinkFailedDialog dialog { potentialTarget != nullptr ?
                                    &ProjectWindow::Get(*potentialTarget) :
-                                   nullptr };
+                                   nullptr,
+                                AudiocomTrace::OpenFromCloudMenu };
```
Comment (grliszas14) [inline modules/sharing/mod-cloud-audiocom/CloudProjectOpenUtils.cpp:167]: nitpick: formatting

File: libraries/lib-cloud-audiocom/sync/ProjectCloudExtension.cpp [hunk 2]
```diff
@@ -437,7 +443,10 @@ void ProjectCloudExtension::OnUpdateSaved(const ProjectSerializer& serializer)
       }
    }
    else
-      Publish({ ProjectSyncStatus::Failed }, false);
+      Publish(
+         { AudiocomTrace::ignore, // TODO Is this correct ?
```
Comment (grliszas14) [inline libraries/lib-cloud-audiocom/sync/ProjectCloudExtension.cpp:447]: should be ok

File: modules/sharing/mod-cloud-audiocom/LinkUrlHandler.cpp [hunk 3]
```diff
@@ -22,7 +23,10 @@ auto subscription = URLSchemesRegistry::Get().Subscribe(
    {
       using namespace audacity::cloud::audiocom;
 
-      if (GetOAuthService().HandleLinkURI(message.url, {}))
+      if (GetOAuthService().HandleLinkURI(
+             message.url,
+             // TODO Is this correct?
```
Comment (grliszas14) [inline modules/sharing/mod-cloud-audiocom/LinkUrlHandler.cpp:28]: this should be good too

**Review by grliszas14 — APPROVED** [review]

File: src/export/ExportAudioDialog.cpp [hunk 4]
```diff
@@ -53,8 +53,9 @@ namespace
 {
 const bool hookRegistered = [] {
    ExportUtils::RegisterExportHook(
-      [](AudacityProject& project, const FileExtension& format, bool selectedOnly)
-      {
+      [](AudacityProject& project, const FileExtension& format,
+         audacity::cloud::audiocom::AudiocomTrace /*ignore*/,
```
Comment (vsverchinsky) [inline src/export/ExportAudioDialog.cpp:57]: Would that build without `lib-cloud-audiocom` dependency?

**Review by vsverchinsky — DISMISSED** [review]
Can't build without `lib-cloud-audiocom`

**Review by LWinterberg — APPROVED** [review]
LGTM from me and also from Lukas and Arif!

**Pull request merged**